src/next_array.rs

```diff
@@ -0,0 +1,80 @@
+use core::mem::MaybeUninit;
+
+/// Helper struct to build up an array element by element.
+struct ArrayBuilder<T, const N: usize> {
+    arr: [MaybeUninit<T>; N],
+    i: usize
+}
+
+impl<T, const N: usize> ArrayBuilder<T, N> {
+    pub fn new() -> Self {
+        Self { arr: maybe_uninit::uninit_array(), i: 0 }
+    }
+    
+    pub unsafe fn push_unchecked(&mut self, x: T) {
```

Comment:
Needs a safety comment, in the format of:
```suggestion
    /// Does XYZ.
    ///
    /// # Safety
    /// 
    /// Callers promises that blah blah blah.
    ///
    /// # Panics
    ///
    /// This method does (or does not) panic.
    pub unsafe fn push_unchecked(&mut self, x: T) {
```